erts/emulator/beam/erl_nif.c

```diff
@@ -4439,6 +4439,16 @@ Eterm erts_load_nif(Process *c_p, ErtsCodePtr I, Eterm filename, Eterm args)
     {
         ErtsStaticNifEntry* sne;
         sne = erts_static_nif_get_nif_init((char*)mod_atomp->name, mod_atomp->len);
+        if (sne == NULL) {
+            /* Second lookup in the static nif table based on the filename */
+            /* this allows Elixir modules which always have dots '.' in their */
+            /* module names to be loaded */
+            char *basename = strrchr(lib_name, '/');
+            if (basename) {
+                basename++;
+                sne = erts_static_nif_get_nif_init(basename, strlen(basename));
+            }
+        }        
```

Comment:
@dominicletz I have a question:
About this fix in `erlang:load_nif` with the second lookup based on the filename. Your comment says it was to work around Elixir modules with '.' in the name. Did you ever got that to work, a statically Elixir NIF module?
I ask because when I try to compile a static NIF module with '.' in the name:

```
#define STATIC_ERLANG_NIF
#include <erl_nif.h>
:
ERL_NIF_INIT(sverker.test, nif_funcs, NULL, NULL, NULL, NULL)
```

It fails because the macro invokation tries to concatenate the module name `sverker.test` and `_nif_init` to form a unique function name. And C functions cannot contain '.' in their names.

Reply:
Yes, the exqlite package can be linked natively now. To make this work there is some c macro magic at the end of the file here: https://github.com/elixir-sqlite/exqlite/blob/2c3138a46331d282f2032574ca4a1897e8e29f5d/c_src/sqlite3_nif.c#L860

It's nothing that I would call a clean solution, but I did not find a better one. Let me know if you have any thoughts on improving the approach.

Reply:
Thank you, @dominicletz. I suspected that you must have done sort of hack to make it work. 

I will post a PR soon to officially support Elixir NIF modules with dots in the module name. And that will also remove the need for this second-lookup-by-filename fix.

Reply:
#5477